test/test.js

```diff
@@ -63,6 +65,25 @@ describe('usb-detection', function() {
 				});
 			});
 
+			it('should return a promise when vid and pid are given', function(done) {
+				usbDetect.find(USB_TEST_VID, USB_TEST_PID)
```

Comment:
Thinking about this more to automate, we could also just pick a device that is already connected instead of having to specify via environment variables `TEST_VID`/`TEST_PID`

Feel free to use the `async`/`await` stuff in the tests,

```js
const devices = await usbDetect.find();

const devicesFromTestedFunction = usbDetect.find(devices[0].vendorId, devices[0].productId);

expect(devicesFromTestedFunction.length).to.be.greaterThan(0);
// Should find a subset of USB devices. We assume you have many USB devices connected
expect(devicesFromTestedFunction.length).to.be.lessThan(devices.length);
```

Reply:
Updated the tests, the question does remain if you should use the thing you test in your tests.

Reply:
We can iterate later if we run into problems